AdaptiveJetStream/jetstream/src/main/java/com/google/jetstream/presentation/screens/videoPlayer/components/ImmersiveModeRequest.kt

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright 2025 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.jetstream.presentation.screens.videoPlayer.components
+
+import android.app.Activity
+import android.os.Build
+import androidx.core.view.WindowCompat
+import androidx.core.view.WindowInsetsCompat
+
+internal fun Activity.toggleImmersiveMode() {
+    if (isImmersive) {
+        requestImmersiveMode(ImmersiveModeRequest.Exit)
+    } else {
+        requestImmersiveMode(ImmersiveModeRequest.Enter)
+    }
+}
+
+internal sealed interface ImmersiveModeRequest {
+
+    fun intoFullScreenModeRequest(): Int
+
+    data object Enter : ImmersiveModeRequest {
+        override fun intoFullScreenModeRequest(): Int {
+            return Activity.FULLSCREEN_MODE_REQUEST_ENTER
+        }
+    }
+
+    data object Exit : ImmersiveModeRequest {
+        override fun intoFullScreenModeRequest(): Int {
+            return Activity.FULLSCREEN_MODE_REQUEST_EXIT
+        }
+    }
+}
+
+private fun Activity.requestImmersiveMode(request: ImmersiveModeRequest) {
+    val insetsController = WindowCompat.getInsetsController(window, window.decorView)
+    when (request) {
+        ImmersiveModeRequest.Enter -> {
+            insetsController.hide(WindowInsetsCompat.Type.systemBars())
+        }
+        else -> {
+            insetsController.show(WindowInsetsCompat.Type.systemBars())
+        }
+    }
+
+    if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.UPSIDE_DOWN_CAKE) {
+        requestFullscreenMode(request.intoFullScreenModeRequest(), null)
+        isImmersive = !isImmersive
```

Comment:
This looks like a toggle, but shouldn't this be based off of `request`?

Reply:
Makes sense. Updated as suggested.